src/parser.cpp

```diff
@@ -724,12 +724,12 @@ void Parser::ParserImpl::loadVariable(const VariablePtr &variable, const XmlNode
 void Parser::ParserImpl::loadConnection(const ModelPtr &model, const XmlNodePtr &node) const
 {
     // Define types for variable and component pairs.
-    using NamePair = std::pair<std::string, std::string>;
+    using NamePair = std::vector<std::string>;
```

Comment:
This typedef doesn't make sense or give the correct information to what it is typedefing.

Reply:
> I don't understand why the component pair was changed from being a pair. It makes good sense that they are a pair because that is what they are not a vector.

It was to reuse the same typedef for the mapping. I can split it up and have four kinds if you'd rather?  Otherwise I need a type for the connection information (with 2 items) and a separate one for the mapping information (with 3 items), and then two separate collections for each of them ... it seemed better this way?

Reply:
Agreed about the naming (I completely missed that). If anything, it should be called `NameInfo`.

Reply:
Compromised in 27563871